oscarapi/middleware.py

```diff
@@ -0,0 +1,167 @@
+import logging
+import re
+
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.core.exceptions import PermissionDenied
+from django.core.urlresolvers import reverse
+from django.http.response import HttpResponse
+from django.utils.translation import ugettext as _
+from rest_framework import exceptions
+from rest_framework import authentication
+
+from oscarapi.utils import (
+    get_domain,
+    session_id_from_parsed_session_uri,
+    get_session
+)
+from oscarapi import models
+
+logger = logging.getLogger(__name__)
```

Comment:
Well, since loggers are organized in a hierarchy, you can easily configure the logger for oscarapi and be done with all the other ones (https://docs.python.org/2/library/logging.html#logging.Logger.propagate). On the contrary, If I would use oscar.api as the logger everywhere, I couldn't enable a debug logger somewhere deep in a package that I am trying to debug.